zio-http/src/main/scala/zio/http/model/headers/values/Allow.scala

```diff
@@ -0,0 +1,36 @@
+package zio.http.model.headers.values
+
+import zio.Chunk
+import zio.http.model.Method
+
+import scala.annotation.tailrec
+
+/**
+ * The Allow header must be sent if the server responds with a 405 Method Not Allowed status code to indicate
+ * which request methods can be used.
+ */
+final case class Allow(methods: Chunk[Method])
+
+object Allow {
+
+  def toAllow(value: String): Allow = {
+    @tailrec def loop(index: Int, value: String, acc: Allow): Allow = {
```

Comment:
Not really sure if we actually want to use recursion here, sounds like an overkill to parse headers to be honest.

Reply:
Replaced it with built-in functions

Reply:
recursion was ok, split and fold are allocating memory

Reply:
reverted the implementation back to the recursive version